### PR TITLE
Run next pylint on AppConfig

### PR DESCRIPTION
**Pull request by YalinLi0312**

### CODE REVIEW

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py [hunk 1]
```diff
@@ -7,12 +7,18 @@
...
+
 from azure.core.pipeline.policies import HTTPPolicy
-from ._utils import get_current_utc_time
+from azure.core.pipeline.transport import AioHttpTransport  # pylint:disable=non-abstract-transport-import,no-name-in-module
 
+from ._utils import get_current_utc_time
 
 class AppConfigRequestsCredentialsPolicy(HTTPPolicy):
-    """Implementation of request-oauthlib except and retry logic."""
+    """Implementation of request-oauthlib except and retry logic.
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py:18]: newline.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py [hunk 2]
```diff
@@ -7,12 +7,18 @@
 import hashlib
 import base64
 import hmac
+from yarl import URL
+
 from azure.core.pipeline.policies import HTTPPolicy
-from ._utils import get_current_utc_time
+from azure.core.pipeline.transport import AioHttpTransport  # pylint:disable=non-abstract-transport-import,no-name-in-module
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py:13]: Why we move it here?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 3]
```diff
@@ -115,28 +115,28 @@ class FeatureFlagConfigurationSetting(ConfigurationSetting):  # pylint: disable=
...
+    :ivar content_type:
+    :vartype content_type: str
     :ivar last_modified:
     :vartype last_modified: datetime
     :ivar read_only:
     :vartype read_only: bool
-    :param tags:
-    :type tags: Dict[str, str]
+    :ivar tags:
+    :vartype tags: Dict[str, str]
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:139]: dict

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 4]
```diff
@@ -115,28 +115,28 @@ class FeatureFlagConfigurationSetting(ConfigurationSetting):  # pylint: disable=
...
-    :ivar feature_id:
-    :vartype feature_id: str
+    :param feature_id:
+    :paramtype feature_id: str
     :ivar value: The value of the configuration setting
     :vartype value: str
     :keyword enabled:
     :paramtype enabled: bool
     :keyword filters:
     :paramtype filters: List[Dict[str, Any]]
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:125]: list[dict[str, Any]]

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 5]
```diff
@@ -115,28 +115,28 @@ class FeatureFlagConfigurationSetting(ConfigurationSetting):  # pylint: disable=
 
     :ivar etag: Entity tag (etag) of the object
     :vartype etag: str
-    :ivar feature_id:
-    :vartype feature_id: str
+    :param feature_id:
+    :paramtype feature_id: str
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:119]: type

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 6]
```diff
@@ -253,21 +253,21 @@ class SecretReferenceConfigurationSetting(ConfigurationSetting):
 
     :ivar etag: Entity tag (etag) of the object
     :vartype etag: str
-    :ivar key:
-    :vartype key: str
-    :ivar secret_id:
-    :vartype secret_id: str
-    :param label:
+    :param key:
+    :paramtype key: str
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:257]: type

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 7]
```diff
@@ -253,21 +253,21 @@ class SecretReferenceConfigurationSetting(ConfigurationSetting):
...
+    :param secret_id:
+    :paramtype secret_id: str
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:259]: type

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 8]
```diff
@@ -253,21 +253,21 @@ class SecretReferenceConfigurationSetting(ConfigurationSetting):
...
+    :ivar label:
     :type label: str
-    :param content_type:
+    :ivar content_type:
     :type content_type: str
     :ivar value: The value of the configuration setting
     :vartype value: Dict[str, Any]
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:265]: dict

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 9]
```diff
@@ -253,21 +253,21 @@ class SecretReferenceConfigurationSetting(ConfigurationSetting):
...
     :ivar last_modified:
     :vartype last_modified: datetime
     :ivar read_only:
     :vartype read_only: bool
-    :param tags:
+    :ivar tags:
     :type tags: Dict[str, str]
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:271]: dict

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_sync_token.py [hunk 10]
```diff
@@ -30,15 +30,19 @@
 
 
 class SyncToken(object):
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_sync_token.py:32]: Do we still need to inherit from object?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py (hunk 2 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py:13]: Just moved imports as others to the start of each file. Any reason to treat them differently?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_sync_token.py (hunk 10 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_sync_token.py:32]: I think it's safe to get rid of.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py (hunk 2 above)
Comment (xiangyan99) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py:13]: The problem is if aiohttp is not installed, it fails the sync client which is not what we want.

Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py:13]: Got it

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py [hunk 11]
```diff
@@ -7,12 +7,27 @@
 import hashlib
 import base64
 import hmac
+
+try:
+    from yarl import URL
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py:12]: We don't need to try them separately.

aiohttp installs yarl.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 12]
```diff
@@ -153,8 +153,8 @@ def list_configuration_settings(
         :param label_filter: filter results based on their label. '*' can be
          used as wildcard in the beginning or end of the filter
         :type label_filter: str
-        :keyword datetime accept_datetime: retrieve ConfigurationSetting existed at this datetime
-        :keyword List[str] fields: specify which fields to include in the results. Leave None to include all fields
+        :keyword str accept_datetime: retrieve ConfigurationSetting existed at this datetime
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:156]: Why this is a string? from the sample code below, seems like this is a datetime object.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 13]
```diff
@@ -216,7 +216,7 @@ def get_configuration_setting(
         :type etag: str or None
         :param match_condition: The match condition to use upon the etag
         :type match_condition: :class:`~azure.core.MatchConditions`
-        :keyword datetime accept_datetime: retrieve ConfigurationSetting existed at this datetime
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:219]: same

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 14]
```diff
@@ -253,22 +253,22 @@ class SecretReferenceConfigurationSetting(ConfigurationSetting):
...
-    :vartype key: str
-    :ivar secret_id:
-    :vartype secret_id: str
-    :param label:
+    :param key:
+    :paramtype key: str
+    :param secret_id:
+    :paramtype secret_id: str
+    :ivar label:
     :type label: str
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:261]: vartype

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 15]
```diff
@@ -253,22 +253,22 @@ class SecretReferenceConfigurationSetting(ConfigurationSetting):
...
-    :param content_type:
+    :ivar content_type:
     :type content_type: str
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:263]: vartype

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py [hunk 16]
```diff
@@ -253,22 +253,22 @@ class SecretReferenceConfigurationSetting(ConfigurationSetting):
...
-    :vartype value: Dict[str, Any]
+    :vartype value: dict[str, Any]
     :ivar last_modified:
     :vartype last_modified: datetime
     :ivar read_only:
     :vartype read_only: bool
-    :param tags:
-    :type tags: Dict[str, str]
+    :ivar tags:
+    :type tags: dict[str, str]
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py:271]: vartype

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 12 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:156]: It is `str`in generated operation: https://github.com/Azure/azure-sdk-for-python/blob/aced1e299c25878d9b96325690564bf6edf89c33/sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_generated/operations/_azure_app_configuration_operations.py#L42

We need to update the sample.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_sync_token.py [hunk 17]
```diff
@@ -29,16 +29,20 @@
 from azure.core.pipeline.policies import SansIOHTTPPolicy
 
 
-class SyncToken(object):
-    """The sync token structure"""
+class SyncToken:
+    """The sync token structure
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_sync_token.py:33]: new line

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**